grpc/src/test/java/io/stargate/grpc/service/MessageStreamObserverTest.java

```diff
@@ -0,0 +1,119 @@
+package io.stargate.grpc.service;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import io.grpc.StatusException;
+import io.grpc.stub.StreamObserver;
+import io.stargate.proto.QueryOuterClass;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiFunction;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+class MessageStreamObserverTest {
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldOnCompleteWhileThereIsNoInFlight() {
+    // given
+    StreamObserver<QueryOuterClass.Response> callerStreamObserver = mock(StreamObserver.class);
+    StreamObserver<QueryOuterClass.Response> responseStreamObserver =
+        new SynchronizedStreamObserver<>(callerStreamObserver);
+    ExceptionHandler exceptionHandler = new ExceptionHandler(responseStreamObserver);
+    BiFunction<QueryOuterClass.Response, AtomicLong, MessageHandler<QueryOuterClass.Response, ?>>
+        handlerProducer =
+            (response, inFlight) -> {
+              inFlight.decrementAndGet(); // inFlight is decremented
+              return mock(MessageHandler.class);
+            };
+    ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+
+    MessageStreamObserver<QueryOuterClass.Response> observer =
+        new MessageStreamObserver<>(
+            responseStreamObserver, exceptionHandler, handlerProducer, executor, 10);
+
+    // when
+    observer.onNext(QueryOuterClass.Response.newBuilder().build());
+
+    // then should immediately complete and call the caller onComplete
+    assertDoesNotThrow(observer::onCompleted);
```

Comment:
what's `assertDoesNotThrow` needed for? if it throws anything, the test will fail